funnel/assets/js/schedule_view.js

```diff
@@ -1,280 +1,287 @@
-            pageTitle: this.get(event.keypath + '.talks.title'),
-            description: this.get(event.keypath + '.talks.speaker') ? this.get(event.keypath + '.talks.title') + ' by ' + this.get(event.keypath + '.talks.speaker') : this.get(event.keypath + '.talks.title') + ", " + this.get('pageDetails')['projectTitle'],
-            url: backPage
-          };
-        } else {
-          pageDetails = {
-            title: activeSession.title + ' – ' + this.get('pageDetails')['projectTitle'],
+          return descriptionHtml.replace(/<img[^>]*>/g, '').substring(0, 400).concat('..');
+        },
+        updateMetaTags(pageDetails) {
+          $('title').html(pageDetails.title);
+          $('meta[name="DC.title"]').attr('content', pageDetails.pageTitle);
+          $('meta[property="og:title"]').attr('content', pageDetails.pageTitle);
+          $('meta[name=description]').attr('content', pageDetails.description);
+          $('meta[property="og:description"]').attr('content', pageDetails.description);
+          $('link[rel=canonical]').attr('href', pageDetails.url);
+          $('meta[property="og:url"]').attr('content', pageDetails.url);
+        },
+        handleBrowserHistory() {
+          // On closing modal, update browser history
+          $('#session-modal').on($.modal.CLOSE, () => {
+            this.modalHtml = '';
+            window.history.pushState('', '', this.pageDetails.url);
+            this.updateMetaTags(this.pageDetails);
+          });
+          // Event listener for back key press since opening modal update browser history
+          $(window).on('popstate', () => {
+            if (this.modalHtml) {
+              $.modal.close();
+            } else if (window.history.state) {
+              // Open the modal with previous session viewed
+              this.openModal(window.history.state.html,
+                window.history.state.backPage, window.history.state.pageDetails);
+            }
+          });
+        },
+        openModal(sessionHtml, backPage, pageDetails) {
+          this.modalHtml = sessionHtml;
+          $('#session-modal').modal('show');
+          window.history.pushState({
+            html: sessionHtml, backpage: backPage, pageDetails,
+          }, '', backPage);
+          this.updateMetaTags(pageDetails);
+        },
+        showSessionModal(activeSession) {
+          const backPage = `${this.pageDetails.url}/${activeSession.url_name_suuid}`;
+          const pageDetails = {
+            title: `${activeSession.title} — ${this.pageDetails.projectTitle}`,
             pageTitle: activeSession.title,
-            description: activeSession.speaker ? activeSession.title + ' by ' + activeSession.speaker : activeSession.title + ", " + this.get('pageDetails')['projectTitle'],
-            url: backPage
+            description: activeSession.speaker ? `${activeSession.title} by ${activeSession.speaker}`
```

Comment:
Translation needed since there's a `by`.

Reply:
To be fixed in https://github.com/hasgeek/funnel/issues/498